database.rules.json

```diff
@@ -1,6 +1,26 @@
 {
   "rules": {
-    ".read": "auth != null",
-    ".write": "auth != null"
-  }
+    "users": {
+      "$uid": {
+        ".read" : "auth !== null && auth.uid === $uid",
+        ".write" : "auth !== null && auth.uid === $uid"
+      }
+    },
+    "calendars":{
+      "$calendarId": {
+        "calendarState": {
+          ".read" : "auth !== null && data.parent().child('read/' + auth.uid).exists()",
+          ".write" : "auth !== null && data.parent().child('write/' + auth.uid).exists()"
+        },
+        "write" : {
```

Comment:
I feel like "write" and "read" here could have better names. Maybe `writers` and `readers` or `can_write` and `can_read`.